asdf/extension.py

```diff
@@ -244,4 +244,10 @@ def reset(self):
         self._extension_list = None
         self._package_metadata = {}
 
+    def resolver(self, uri):
+        tag_mapping = self.extension_list.tag_mapping
+        url_mapping = self.extension_list.url_mapping
+        return url_mapping(tag_mapping(uri))
```

Comment:
Nice.  This is potentially useful elsewhere to get the full-bells-and-whistles resolver so that we don't have to instantiate a dummy `AsdfFile` object just to get to the resolver.